### PR TITLE
prodsync: run R/O on slave DB

### PR DESCRIPTION
**Pull request by kaplun**
Signed-off-by: Samuele Kaplun <samuele.kaplun@cern.ch>

### CODE REVIEW

**Comment by kaplun** [comment]
@tsgit WDYT?

File: bibtasklets/bst_prodsync.py [hunk 1]
```diff
@@ -96,6 +114,8 @@ def redis_sync(modified_records, time_estimator, tot):
     """Sync to redis."""
     r = redis.StrictRedis.from_url(CFG_REDIS_HOST_LABS)
     for i, recid in enumerate(modified_records):
+        with run_ro_on_slave_db():
+            record = format_record(recid, 'xme', user_info=ADMIN_USER_INFO)[0]
         record = format_record(recid, 'xme', user_info=ADMIN_USER_INFO)[0]
```
Comment (david-caro) [inline bibtasklets/bst_prodsync.py]: shouldn't this line disappear?

Comment (kaplun) [inline bibtasklets/bst_prodsync.py]: Totally :+1:

File: bibtasklets/bst_dump_records.py [hunk 2]
```diff
@@ -35,6 +35,26 @@
...
+
+class run_ro_on_slave_db:
+    """
+    Force the usage of the slave DB in read-only mode
+    """
+    def __enter__(self):
+        from invenio import dbquery_config
+        from invenio import dbquery
+        self.old_host = dbquery_config.CFG_DATABASE_HOST
+        dbquery_config.CFG_DATABASE_HOST = dbquer_config.CFG_DATABASE_SLAVE
```
Comment (tsgit) [inline bibtasklets/bst_dump_records.py]: typo: `dbquer_config.`...

File: bibtasklets/bst_prodsync.py [hunk 3]
```diff
@@ -53,6 +53,24 @@
...
 
+class run_ro_on_slave_db:
+    """
+    Force the usage of the slave DB in read-only mode
+    """
+    def __enter__(self):
+        from invenio import dbquery_config
+        from invenio import dbquery
+        self.old_host = dbquery_config.CFG_DATABASE_HOST
+        dbquery_config.CFG_DATABASE_HOST = dbquer_config.CFG_DATABASE_SLAVE
```
Comment (tsgit) [inline bibtasklets/bst_prodsync.py]: same typo as above `dbquer_` instead of `dbquery_`

File: bibtasklets/bst_prodsync.py (hunk 1 above)
Comment (tsgit) [inline bibtasklets/bst_prodsync.py]: yes, this line must disappear

**Pull request merged**